core/src/main/java/io/questdb/griffin/engine/functions/rnd/RndStringMemory.java

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.rnd;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.vm.Vm;
+import io.questdb.cairo.vm.api.MemoryAR;
+import io.questdb.griffin.SqlException;
+import io.questdb.std.MemoryTag;
+import io.questdb.std.Misc;
+import io.questdb.std.Rnd;
+
+import java.io.Closeable;
+
+class RndStringMemory implements Closeable {
+    private final MemoryAR strMem;
+    private final MemoryAR idxMem;
+    private final int count;
+    private final int lo;
+    private final int hi;
+
+    RndStringMemory(String signature, int count, int lo, int hi, int position, CairoConfiguration configuration) throws SqlException {
+        this.count = count;
+        this.lo = lo;
+        this.hi = hi;
+
+        final int pageSize = configuration.getRndFunctionMemoryPageSize();
+        final int maxPages = configuration.getRndFunctionMemoryMaxPages();
+        final long actualMem = (long) maxPages * pageSize;
+        // check against worst case, the highest possible mem usage
+        final long hiMem = count * (Vm.getStorageLength(hi) + Long.BYTES);
```

Comment:
Please consider adding reference to cairo.rnd.memory.* entries to error message to help user fix the problem if settings are too low .

Reply:
> Please consider adding reference to cairo.rnd.memory.* entries to error message to help user fix the problem if settings are too low .

not a bad idea but i have not really seen examples of this.
it seems to me that we usually keep the error messages short and do not provide lengthy explanations or hints how to fix the issue.
might be something we want to do in the future.
anyone else has a firm opinion on this topic?

however, what i am sure about is that a description should make it into our doc.
the config part and the page talks about rnd functions should explain what these settings are.
will update the doc definitely.

Reply:
Doc-wise I agree . 
As for error message - it would be more useful to the user if we either explained what to change OR added some kind of  
error code .